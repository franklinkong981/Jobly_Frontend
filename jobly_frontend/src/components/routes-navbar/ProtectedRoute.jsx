import React,{useContext} from "react";
import {Route, Navigate} from "react-router-dom";
import currentUserContext from "../contexts-currentUserContext.jsx";

function ProtectedRoute({path, element}) {
  const {currentUserInfo} = useContext(currentUserContext);

  if (!currentUserInfo) {
    return <Navigate to="/login" />
  }

  return (
    <Route exact path={path} element={element} />
  );
}

export default ProtectedRoute;