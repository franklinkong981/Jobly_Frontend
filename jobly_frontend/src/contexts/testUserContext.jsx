import React from "react";
import CurrentUserContext from "./currentUserContext.jsx";

const testUser = {
  username: "testuser",
  first_name: "Testy",
  last_name: "Usery",
  email: "testuser981@gmail.com"
};

const TestUserProvider = ({children, currentTestUser = testUser, hasUserAppliedToJob= () => false}) => (
  <CurrentUserContext.Provider value={{currentTestUser, hasUserAppliedToJob}}>
    {children}
  </CurrentUserContext.Provider>
);

export {TestUserProvider};