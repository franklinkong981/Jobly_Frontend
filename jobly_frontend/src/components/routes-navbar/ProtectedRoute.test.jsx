import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import ProtectedRoute from "./ProtectedRoute.jsx";

import {TestUserProvider} from "../../contexts/testUserContext.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <TestUserProvider>
        <ProtectedRoute />
      </TestUserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot when logged in", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider>
        <ProtectedRoute />
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider currentTestUser={null}>
        <ProtectedRoute />
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});