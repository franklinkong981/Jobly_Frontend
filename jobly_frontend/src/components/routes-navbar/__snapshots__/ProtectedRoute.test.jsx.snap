// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot when logged in 1`] = `<DocumentFragment />`;

exports[`matches snapshot when logged out 1`] = `<DocumentFragment />`;
